### PR TITLE
Fix dump tools premultiplied alpha.

### LINKED ISSUE #11967: Export PNG from serializers without losing color information due to premultiplied alpha
See https://forum.babylonjs.com/t/exported-gltf-scene-does-not-have-material-color-in-blender/27235/27 for context.

The gist of the problem is that color information with non-1 alpha doesn't export properly because the code currently uses canvas to get the PNG and the canvas premultiplies alpha.

I found a way to use offscreen canvas, image bitmap, transferFromImageBitmap, and convertToBlob that appears to keep the color information intact. It also doesn't translate from and to base64 which should mean it will be faster.

https://jsfiddle.net/k63b2vL7/

Question: Does this work on all relevant browsers?

### PR DESCRIPTION
**Pull request by sebavan**
Fix #11967

Fixes as well the generation of Environment Textures.

### CODE REVIEW

**Review by bghgary — APPROVED** [review]
LGTM

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts [hunk 1]
```diff
@@ -22,6 +21,7 @@ import "../../Engines/Extensions/engine.renderTarget";
 import "../../Engines/Extensions/engine.renderTargetCube";
 import { Engine } from "../../Engines/engine";
 import { _ObserveArray } from "core/Misc/arrayTools";
+import { DumpTools } from "core/Misc/dumpTools";
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Textures/renderTargetTexture.ts:24]: Shouldn't this import (and the one above) be relative?

File: packages/dev/core/src/Misc/tools.ts [hunk 2]
```diff
@@ -842,25 +766,25 @@ export class Tools {
...
      * Encodes the canvas data to base 64 or automatically download the result if filename is defined
+     * @param canvas canvas to get the data from.
      * @param successCallback defines the callback triggered once the data are available
      * @param mimeType defines the mime type of the result
      * @param fileName defines he filename to download. If present, the result will automatically be downloaded
-     * @param canvas canvas to get the data from. If not provided, use the default screenshot canvas
      * @param quality defines the quality of the result
      */
     static EncodeScreenshotCanvasData(
+        canvas: HTMLCanvasElement,
```
Comment (Popov72) [inline packages/dev/core/src/Misc/tools.ts:776]: This one is a breaking change? Though I don't know if a lot of people are calling `EncodeScreenshotCanvasData` directly...

**Review by Popov72 — APPROVED** [review]
LGTM, only small comments.

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts (hunk 1 above)
Comment (RaananW) [reply packages/dev/core/src/Materials/Textures/renderTargetTexture.ts:24]: I made a change to support package-based imports for different use-cases, but yes, it is always better to keep it relative anyhow.

File: packages/dev/core/src/Misc/tools.ts (hunk 2 above)
Comment (RaananW) [reply packages/dev/core/src/Misc/tools.ts:776]: It is. Was there a need to change the signature?

**Comment by RaananW** [comment]
Visual tests (both webgl1 and 2) fail on `GLTF Serializer KHR materials clearcoat` with a timeout. I don't see any exception reported TBH, so I am not entirely sure why

File: packages/dev/core/src/Misc/tools.ts (hunk 2 above)
Comment (sebavan) [reply packages/dev/core/src/Misc/tools.ts:776]: The other function was taking a temp canvas by default which could lead to a lot of weird side effects :-) I ll change it back if the community asks for it but pretty sure never used externally

**Pull request merged**